src/BTCPayServer.Lightning.LND/LndClient.cs

```diff
@@ -279,7 +279,7 @@ internal static LightningInvoice ConvertLndInvoice(LnrpcInvoice resp)
                 Status = LightningInvoiceStatus.Unpaid
             };
             invoice.ExpiresAt = DateTimeOffset.FromUnixTimeSeconds(ConvertInv.ToInt64(resp.Creation_date) + ConvertInv.ToInt64(resp.Expiry));
-            if (resp.Settle_date != null)
+            if (resp.Settled == true)
```

Comment:
Cool, yeah this is much more reliable. Looking at the existing code, I guess since the `Settle_date` was parsed as a string (even though it's an int), when we started to always populate all fields, it was picked up a non-null, even though it was zero (invoice was never settled). 

If possible can you also try with 0.10-rc5? I think it should work np. 0.10 is great for btcpay as with this release both the sender and the receiver now support multi-part payments.

Reply:
> If possible can you also try with 0.10-rc5?

Tests pass with 0.9.2, but not with 0.10.0. I'm in the middle of some other coding, so I'll come to this tomorrow. I hope you'll monitor issue in case I can't resolve it on my own. Thanks for the help.

Reply:
Does `Settled` exists in old version of LND? If it does not, should be a `bool?`, and we should fallback to `Settle_date` to not break with previous versions.

Reply:
@NicolasDorier `Settled` exists in old versions. I haven't regenerated that Swagger REST API wrapper since 2018 because of support for old versions. But we should definitely start considering obsoleting some of the LND versions (especially those that are not in the wild because of vulnerability).

Reply:
Yeah `Settled` has been around since `lnd` has been on mainnet, it actually predates `Settle_date`.